catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -465,6 +474,23 @@ private void skipBytes(InputStream is, String bytesToSkip) throws IOException {
         }
     }
 
+    private void alignStream(InputStream in, long requestedBytesToSkip, long responseBytesSkipped)
+            throws IOException {
```

Comment:
Unfortunately, yes. If Partial Content responses are not supported by the server, then it will return the entire content, in which case we need to skip ahead in the input stream to avoid re-reading bytes. It is also possible for the server to support Partial Content responses and return a different byte range than what was requested. If we ask for bytes 5-10, for whatever reason, the server _is_ allowed to return a partial content response with the range 4-10, and we would have to skip over the 1 overlapping byte. The server *shouldn't* return a byte range higher than 5-10 (like 6-10), but I added that situation just in case to be thorough.